isis/src/base/objs/Camera/Camera.cpp

```diff
@@ -579,23 +579,36 @@ namespace Isis {
    *
    * @return @b double
    */
-  double Camera::ObliqueDetectorResolution(){
+  double Camera::ObliqueDetectorResolution(bool useLocal) {
 
 
-      if(HasSurfaceIntersection()){
+    if(!HasSurfaceIntersection()) {
+      return -1.0;
+    }
 
-          double thetaRad;
-          thetaRad = EmissionAngle()*DEG2RAD;
+    double thetaRad;
+    double emissionDeg;
 
-          if (thetaRad < HALFPI) {
-            return DetectorResolution()/cos(thetaRad);
+    if(useLocal) {
+      Angle phase, emission, incidence;
+      bool success;
 
-          }
-          return Isis::Null;
+      LocalPhotometricAngles(phase, incidence, emission, success);
+      emissionDeg = (success) ? emission.degrees() : EmissionAngle();
+    }
+    else {
```

Comment:
I would argue that you return Null when the photometric angles cannot be computed. Using the ellipsoid (EmissionAngle()) when local photometric angles fail creates inconsistent geometry. For example, when using local DTMs, there may be areas within an image that does not provide full DTM coverage and in those cases, Nulls are returned for local angles in apps like phocube. If you switch up and essentially ignore the newly added bool argument, you get inconsistent backplanes/data and may need to mask the oblique detector plane or risk misinterpreting results.

In fact, I am not sure under what conditions you would want angles off the ellipsoid when a DTM is available. However, whatever the reason, I see no way for users to specify which one should be used other than explicit shape model management using spiceinit (I added both options to phocube and formally compute the ellipsoid oblique angles for posterity).